homeassistant/components/upnp/config_flow.py

```diff
@@ -294,14 +294,22 @@ def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
 
     async def async_step_init(self, user_input: Mapping = None) -> FlowResult:
         """Manage the options."""
+        LOGGER.debug(
+            "UpnpOptionsFlowHandler.async_step_init: user_input: %s", user_input
+        )
         if user_input is not None:
-            coordinator = self.hass.data[DOMAIN][self.config_entry.entry_id]
             update_interval_sec = user_input.get(
                 CONFIG_ENTRY_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
             )
             update_interval = timedelta(seconds=update_interval_sec)
-            LOGGER.debug("Updating coordinator, update_interval: %s", update_interval)
-            coordinator.update_interval = update_interval
+
+            if self.config_entry.entry_id in self.hass.data[DOMAIN]:
+                coordinator = self.hass.data[DOMAIN][self.config_entry.entry_id]
+                LOGGER.debug(
+                    "Updating coordinator, update_interval: %s", update_interval
+                )
+                coordinator.update_interval = update_interval
```

Comment:
Ideally, we should completely remove the scan interval setting.
We offer the `homeassistant.update_entity` as an alternative.

Reply:
Does the `homeassistant.update_entity` service allow one to decrease the poll rate?

Reply:
What you want it to be.  You can call it automation, how often that triggers, or which patterns, maybe skip Sundays and poll slower on a workday. That is up to the implementer. The services offer more freedom.

Reply:
> I'm not following that comment? What do you mean by loose? If flexible: Yes. That was the goal.

The way you bring it somebody can determine it should be polled once on a sunday, perhaps twice on a monday, and once when easter and pentecost are on the same day. This will prevent some of the sensors from working correctly.

> All of them, it built into the core itself.

Thus this integration is also doing it already?

Reply:
Joking aside, I've removed the options flow.

Reply:
> The way you bring it somebody can determine it should be polled once on a sunday, perhaps twice on a monday, and once when easter and pentecost are on the same day. This will prevent some of the sensors from working correctly.

Well, we can make that more practical, for example, poll my electric vehicle more often during the day than during the night, so I don't wake up my Tesla in the night and thus waste energy.

Or: Reducing polling on a paid cloud service when I'm not home or during nights, to save API calls and thus costs.

We've seen many variations on these, in many feature requests throughout the years.